nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala

```diff
@@ -1,411 +1,472 @@
+
+    else if (fromTypeId == arrayBooleanTypeId && toTypeId == arrayBooleanTypeId) 
+      from.asInstanceOf[BooleanArray].copyToArray (fromPos, to.asInstanceOf[BooleanArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayCharTypeId && toTypeId == arrayCharTypeId) 
+      from.asInstanceOf[CharArray].copyToArray (fromPos, to.asInstanceOf[CharArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayByteTypeId && toTypeId == arrayByteTypeId) 
+      from.asInstanceOf[ByteArray].copyToArray (fromPos, to.asInstanceOf[ByteArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayShortTypeId && toTypeId == arrayShortTypeId) 
+      from.asInstanceOf[ShortArray].copyToArray (fromPos, to.asInstanceOf[ShortArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayIntTypeId && toTypeId == arrayIntTypeId) 
+      from.asInstanceOf[IntArray].copyToArray (fromPos, to.asInstanceOf[IntArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayLongTypeId && toTypeId == arrayLongTypeId) 
+      from.asInstanceOf[LongArray].copyToArray (fromPos, to.asInstanceOf[LongArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayFloatTypeId && toTypeId == arrayFloatTypeId) 
+      from.asInstanceOf[FloatArray].copyToArray (fromPos, to.asInstanceOf[FloatArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 63)
+
+    else if (fromTypeId == arrayDoubleTypeId && toTypeId == arrayDoubleTypeId) 
+      from.asInstanceOf[DoubleArray].copyToArray (fromPos, to.asInstanceOf[DoubleArray], toPos, length)
+      
+// ###sourceLocation(file: "/home/francois/proyectos/oss/scala-native-fbd/scala-native/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala.gyb", line: 68)
+
+    else
+      throw new ArrayStoreException ("Not implemented yet.")
```

Comment:
I think message here is confusing. All the Scala-relevant semantics is already implemented. I think it's best not provide any error message here.